hedyorg/hedy
# [UI] Upload of a CSV file to create student accounts for a class

Work In Progress
This hopefully should give an idea of the user experience.  

Items outstanding
* better UI styling (would love help from someone more visually inclined than me)
* error handling
* content (need to understand proper way to add english and request translations)
* testing

Fixes #3936 
resolves discussion #3933

**How to test**
TBD, ideally cypress, got to figure out how to use it
For now you can try it out manually
* For Teachers > Select a class > Add Students 
* press the new upload_csv button
* drag and drop or select a csv file to upload (csv file must only contain two columns with headers "username" and "password")
* confirm you want to upload the file
* review the changes on the create accounts form
* from here it works as if the accounts were typed in manually.
do all the things you would do if you were going to add them manually, e.g. generate passwords, append classname etc.


![01 new upload csv button](https://user-images.githubusercontent.com/6020902/215687794-5cd3015e-ffaf-4cbf-ad62-f368c9e26497.png)
![02 drag and drop or choose csv file](https://user-images.githubusercontent.com/6020902/215687796-f0f63643-c3d1-4eaf-ba83-c627bc30566e.png)
![03 confirm you want to upload csv](https://user-images.githubusercontent.com/6020902/215687797-27ca5cbb-2e93-4ef0-ae47-2a450f5e0547.png)
![04 preview accounts imported from csv](https://user-images.githubusercontent.com/6020902/215687799-d4f700ea-a130-456c-bcc8-bcd60a021e13.png)
